src/processor/operator/persistent/reader/csv/driver.cpp

```diff
@@ -239,5 +240,64 @@ bool SniffCSVNameAndTypeDriver::addValue(uint64_t rowNum, common::column_id_t co
     return true;
 }
 
+SniffCSVHeaderDriver::SniffCSVHeaderDriver(SerialCSVReader* reader,
+    const function::ScanTableFuncBindInput* /*bindInput*/,
+    const std::vector<std::pair<std::string, common::LogicalType>>& typeDetected)
+    : SerialParsingDriver(getDummyDataChunk(), reader, DriverType::SNIFF_CSV_HEADER) {
+    for (auto i = 0u; i < typeDetected.size(); i++) {
+        columns.push_back({typeDetected[i].first, typeDetected[i].second.copy()});
+    }
+}
+
+bool SniffCSVHeaderDriver::done(uint64_t rowNum) const {
```

Comment:
You can probably make this funciton inline.

Reply:
changed to inline